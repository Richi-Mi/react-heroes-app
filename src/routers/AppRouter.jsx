import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import LoginScreen from '../components/login/LoginScreen';
import DashBoardRoutes from './DashBoardRoutes';
import PrivateRoutes from './PrivateRoutes';
import PublicRoutes from './PublicRoutes'
// import HeroesApp from '../HeroesApp';

const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={
                    <PublicRoutes>
                        <LoginScreen />
                    </PublicRoutes>
                } />

                <Route path='/*' element = { 
                    <PrivateRoutes>
                        <DashBoardRoutes />
                    </PrivateRoutes>
                } />
                {/* <Route path='/*' element = { <DashBoardRoutes /> } /> */}
            </Routes>
        </BrowserRouter>
    )
};

export default AppRouter;
