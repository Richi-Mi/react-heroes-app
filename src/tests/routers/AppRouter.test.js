import { mount } from "enzyme";
import AuthContext from "../../auth/authContext";
import AppRouter from "../../routers/AppRouter";

describe(' * Pruebas sobre el componente <AppRouter /> ', () => {
    
    test(' - Debe de mostrar el Login si no esta autenticado', () => {
        const contextValue = {
            user: {
                logged: false
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe( "Login Screen" );
    });
    
    test(' - Debe mostrar el componente Home si esta autenticado', () => {
        const contextValue = {
            user: {
                logged: true,
                name: 'Ricardo'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.title_home').text().trim() ).toBe('Heroes App');
        expect( wrapper.find('.nav').exists() ).toBe( true );
    });

});
