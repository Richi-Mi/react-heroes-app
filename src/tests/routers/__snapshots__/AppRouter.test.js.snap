// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas sobre el componente <AppRouter />   - Debe de mostrar el Login si no esta autenticado 1`] = `
<div
  className="container_login"
>
  <h1>
     Login Screen 
  </h1>
  <hr />
  <div
    className="form_search"
  >
    <input
      autoComplete="off"
      className="form_input"
      name="username"
      onChange={[Function]}
      placeholder="Your Name"
      type="text"
      value=""
    />
    <button
      className="btn_search btn_login"
      onClick={[Function]}
    >
       Login 
    </button>
  </div>
</div>
`;

exports[` * Pruebas sobre el componente <AppRouter />   - Debe mostrar el componente Home si esta autenticado 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="nav_options"
    >
      <a
        aria-current="page"
        className="nav_link active"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home"
        />
        Inicio
      </a>
      <a
        className="nav_link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav_link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav_link"
        href="/search"
        onClick={[Function]}
      >
        Search
        <i
          className="fas fa-search"
        />
      </a>
    </div>
    <div
      className="nav_acount"
    >
      <span
        className="text_info acount_item"
      >
        Ricardo
      </span>
      <button
        className="btn_logout acount_item"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="content_home animate__animated animate__fadeIn"
    >
      <h2
        className="title_home"
      >
         Heroes App 
      </h2>
      <h3
        className="subtitle_home"
      >
         A list Heroes of Comics 
      </h3>
      <div
        className="gif_home"
      >
        <img
          alt="heroes"
          src="/banner_code.jpg"
        />
      </div>
      <div
        className="btns"
      >
        <a
          className="link_more"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel Heroes
        </a>
        <a
          className="link_more"
          href="/dc"
          onClick={[Function]}
        >
          DC Heroes
        </a>
      </div>
    </div>
  </div>,
]
`;
